doc/source/dsintro.rst

```diff
@@ -822,6 +859,35 @@ completion mechanism so they can be tab-completed:
     In [5]: df.fo<TAB>
     df.foo1  df.foo2
 
+.. _dsintro.data_type:
```

Comment:
note there are a number of sections on dtypes in basics. This is ok here, but at the very least add a cross reference there (or maybe combine and move here). *maybe* move to a top-level pages on Data Types.